### ISSUE TITLE
olevba - parsing error in Mac file

### ISSUE THREAD

**Issue by decalage2** [decalage2/oletools#132]
Issue reported by @noarfromspace: 

`Unknown format code 'X' for object of type 'str'` while running olevba on https://www.virustotal.com/en/file/07adb8253ccc6fee20940de04c1bf4a54a4455525b2ac33f9c95713a8a102f3d/analysis/1486466812/

It looks like it is triggered by an incorrect format string in `__init__` line 391.

**Comment by decalage2** [comment]
This bug was partly fixed by PR #128. But there is still a parsing error that needs to be addressed for that file:

`Unexpected value in VBA/dir for variable REFERENCE_Reserved: expected (003E,000D) but found 000E!`

**Comment by decalage2** [comment]
This issue is fixed by commit 82c4a54. The fix works fine but it is not fully satisfying, because it is just ignoring the parsing errors related to the dir stream, and falls back to the oledump-style heuristics to find macros.

**Comment by c1fe** [comment]
I ran into the same issue. I spent some time looking into it and think that contrary to the specification, the REFERENCENAME record does not always have the Reserved, SizeOfNameUnicode and NameUnicode fields, at least on documents authors on a Mac. 

Once this discrepancy is taken into account, it appears that these files can be parsed according to the specification. 

Here are my changes if you're curious: https://github.com/c1fe/oletools/commit/b8f6d7bc8ff7daee6381477988774f1f6cdf6cfe
